src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -886,9 +923,49 @@ begin
     rwa ← ennreal.of_real_to_real inf_edist_lt_top.ne at key, },
 end
 
-lemma closure_eq_Inter_cthickening {E : set α} :
+lemma cthickening_eq_Inter_thickening {δ : ℝ} (δ_nn : 0 ≤ δ) (E : set α) :
+  cthickening δ E = ⋂ (ε : ℝ) (h : δ < ε), thickening ε E :=
+begin
+  apply le_antisymm,
+  { exact subset_bInter (λ _ hε, cthickening_subset_thickening' (δ_nn.trans_lt hε) hε E), },
+  { rw cthickening_eq_Inter_cthickening δ_nn E,
+    apply bInter_mono,
+    exact λ ε _, thickening_subset_cthickening ε E, },
+end
+
+/-- The closure of a set equals the intersection of its closed thickenings of positive radii. -/
+lemma closure_eq_Inter_cthickening (E : set α) :
   closure E = ⋂ (δ : ℝ) (h : 0 < δ), cthickening δ E :=
-by { rw ← cthickening_zero, exact cthickening_eq_Inter_cthickening rfl.ge, }
+by { rw ← cthickening_zero, exact cthickening_eq_Inter_cthickening rfl.ge E, }
+
+/-- The closure of a set equals the intersection of its (open) thickenings of positive radii. -/
+lemma closure_eq_Inter_thickening (E : set α) :
+  closure E = ⋂ (δ : ℝ) (h : 0 < δ), thickening δ E :=
+by { rw ← cthickening_zero, exact cthickening_eq_Inter_thickening rfl.ge E, }
+
+/-- The frontier of the (open) thickening of a set is contained in an `inf_edist` level set. -/
+lemma frontier_thickening_subset (E : set α) {δ : ℝ} (δ_pos : 0 < δ) :
+  frontier (thickening δ E) ⊆ {x : α | inf_edist x E = ennreal.of_real δ } :=
+begin
+  have singleton_preim :
+    {x : α | inf_edist x E = ennreal.of_real δ } = (λ x , inf_edist x E) ⁻¹' {ennreal.of_real δ},
+  by refl,
+  rw [thickening_eq_preimage_inf_edist, singleton_preim,
+      ← (frontier_Iio' ⟨(0 : ℝ≥0∞), ennreal.of_real_pos.mpr δ_pos⟩)],
+  apply continuous_inf_edist.frontier_preimage_subset (Iio (ennreal.of_real δ)),
```

Comment:
Could you replace that `apply` by `exact`, since it closes the goal?

Reply:
Thank you! Done.